src/map.cpp

```diff
@@ -122,43 +122,35 @@ MapSector * Map::getSectorNoGenerate(v2s16 p)
 	return getSectorNoGenerateNoLock(p);
 }
 
-MapBlock * Map::getBlockNoCreateNoEx(v3s16 p3d)
+MapBlock * Map::getBlockNoCreate(v3s16 p3d)
 {
 	v2s16 p2d(p3d.X, p3d.Z);
 	MapSector * sector = getSectorNoGenerate(p2d);
 	if(sector == NULL)
 		return NULL;
-	MapBlock *block = sector->getBlockNoCreateNoEx(p3d.Y);
-	return block;
-}
-
-MapBlock * Map::getBlockNoCreate(v3s16 p3d)
-{
-	MapBlock *block = getBlockNoCreateNoEx(p3d);
-	if(block == NULL)
-		throw InvalidPositionException();
+	MapBlock *block = sector->getBlockNoCreate(p3d.Y);
 	return block;
 }
 
 bool Map::isNodeUnderground(v3s16 p)
 {
 	v3s16 blockpos = getNodeBlockPos(p);
-	MapBlock *block = getBlockNoCreateNoEx(blockpos);
+	MapBlock *block = getBlockNoCreate(blockpos);
 	return block && block->getIsUnderground(); 
 }
 
 bool Map::isValidPosition(v3s16 p)
 {
 	v3s16 blockpos = getNodeBlockPos(p);
-	MapBlock *block = getBlockNoCreateNoEx(blockpos);
+	MapBlock *block = getBlockNoCreate(blockpos);
 	return (block != NULL);
```

Comment:
This method can be condensed to one line:

```cpp
bool Map::isValidPosition(v3s16 p)
{
	return getBlockNoCreate(getNodeBlockPos(p));
}
```

If I'm not mistaken, this method should return `true` if the returned expression is anything other than `nullptr`, and `false` if `nullptr`.